### PR TITLE
Add qc metrics

### PR DESCRIPTION
**Pull request by AntoniaChroni**
## Description

This PR includes a `README.md` file for the QC metrics to be included during pre-processing of the sequencing library for the 10x workflow. There are two example scripts for the Seurat QC `01_Seurat_QC.Rmd` and doublets estimation `03_run_scDblFinder_non_hashed.Rmd`.

@migbro has made a tool for Seurat QC, other tools mentioned above will be part of later tickets
Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test run with standard test set: https://cavatica.sbgenomics.com/u/d3b-bixu/kf-dev-single-cell-rna/tasks/6c98c540-2197-4f4d-b6f1-116edf09e831/
- [ ] Test B

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have committed any related changes to the PR

### CODE REVIEW

**Comment by AntoniaChroni** [comment]
@migbro scootr can be cloned from [here](https://github.com/igordot/scooter)

**Comment by migbro** [comment]
One quick note for reviewers, aside from the Seurat qc, I have not tested out the other notebooks. It may seem weird as part of the dev process, but you can probably ignore the others as those scripts will be more thoroughly vetted as part of the next ticket to make them into cwl tools

File: tools/cellranger_mkref.cwl [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        }
+        return cmd;
+     }
+  - position: 1
+    shellQuote: false
+    valueFrom: >-
+     ${
+        var cmd = "";
+        if (inputs.attribute != null){
+            cmd = " && cellranger mkref --genes=" + inputs.genes.nameroot + ".filtered.gtf";;
```
Comment (dmiller15) [inline tools/cellranger_mkref.cwl:34]: Double `;`

File: tools/cellranger_mkref.cwl [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        }
+        else{
+            cmd = " cellranger mkref --genes=" + inputs.genes.path;
```
Comment (dmiller15) [inline tools/cellranger_mkref.cwl:37]: Do we need the leading space?

File: tools/cellranger_mkref.cwl [hunk 3]
```diff
@@ -0,0 +1,60 @@
+cwlVersion: v1.2
+class: CommandLineTool
+id: cellranger_mkref
+doc: "Make a custom reference for cellranger"
```
Comment (dmiller15) [inline tools/cellranger_mkref.cwl:4]: Might want to make it clear what this tool does functionally. Explain how providing the `attribute` field changes the behavior of this tool.

File: tools/cellranger_count.cwl [hunk 4]
```diff
@@ -1,12 +1,12 @@
...
+cwlVersion: v1.2
 class: CommandLineTool
 id: cellranger_count
 doc: "Run cellranger count on a set of fastq files"
 
 requirements:
   - class: ShellCommandRequirement
   - class: DockerRequirement
-    dockerPull: 'pgc-images.sbgenomics.com/d3b-bixu/cellranger:6.0'
+    dockerPull: 'pgc-images.sbgenomics.com/brownm28/cellranger:6.1.2'
```
Comment (dmiller15) [inline tools/cellranger_count.cwl:9]: Should this be d3b-bixu? You have it for mkref

**Review by sickler-alex — APPROVED** [review]

**Review by dmiller15 — APPROVED** [review]

**Pull request merged**